### PR TITLE
Missing checks and other improvements

### PR DESCRIPTION
**Pull request by cygnusv**
### What this does:

1. Fixes missing checks in `KFrag.verify()` and in `check_challenge()`
2. Authenticates the original components of the capsule with the AEAD
3. Removed KFrag consistency checks (#77)
4. Fixed some conceptual errors in the user story of the `using_pyumbral.rst` example

### CODE REVIEW

File: docs/source/using_pyumbral.rst [hunk 1]
```diff
@@ -70,25 +70,25 @@ Threshold split-key re-encryption
...
 
 
-Alice generates kfrags for Bob
--------------------------------
-When Alice wants to send a re-encrypted message to Bob,
-*threshold split re-encryption keys*, or *"kfrags"*, are created for
-distribution and later reconstruction via "Shamir's Secret Sharing".
+Alice grants access to Bob by generating kfrags 
+-----------------------------------------------
+When Alice wants to grant Bob access to open her encrypted messages,
```
Comment (jMyles) [inline docs/source/using_pyumbral.rst]: Yeah, nice change.  Very nice.

File: docs/source/using_pyumbral.rst [hunk 2]
```diff
@@ -107,7 +107,7 @@ S3, IPFS, Google Cloud, Sneakernet, etc.
 Bob fails to open the capsule
 -------------------------------
 If Bob attempts to open a capsule that was not encrypted for his public key,
-or re-encrypted for him by Ursula, He will not be able to open it.
+or re-encrypted for him by Ursula, he will not be able to open it.
```
Comment (jMyles) [inline docs/source/using_pyumbral.rst]: I take it you didn't like the religious overtones?

File: docs/source/using_pyumbral.rst [hunk 3]
```diff
@@ -120,44 +120,35 @@ or re-encrypted for him by Ursula, He will not be able to open it.
...
+Bob asks several Ursulas to re-encrypt the capsule so he can open it. 
+Each Ursula performs re-encryption on the capsule using the `kfrag` 
+provided by Alice, obtaining this way a "capsule fragment", or `cfrag`,
+Let's mock a network or transport layer by sampling `M` random `kfrags`,
+one for each required Ursula.
 
-Ursula exchanges Bob's `kfrags` for "capsule fragments", or `cfrags`,
-performing re-encryption with the capsule.
+Bob collects the resulting `cfrags` from several Ursulas. 
+Bob must gather at least `M` `cfrags` in order to activate the capsule.
```
Comment (jMyles) [inline docs/source/using_pyumbral.rst]: Great fixes.

File: umbral/pre.py [hunk 4]
```diff
@@ -85,11 +85,14 @@ def from_bytes(cls, capsule_bytes: bytes, curve: ec.EllipticCurve = None):
         return cls(point_eph_e=eph_e, point_eph_v=eph_v, bn_sig=sig,
                    e_prime=e_prime, v_prime=v_prime, noninteractive_point=eph_ni)
 
+    def _original_to_bytes(self) -> bytes:
```
Comment (jMyles) [inline umbral/pre.py]: Look at the private methods on @cygnusv!  That's right, the `bytes().join()`!

File: umbral/pre.py [hunk 5]
```diff
@@ -482,10 +486,14 @@ def decrypt(capsule: Capsule, priv_key: UmbralPrivateKey,
...
-        cleartext = dem.decrypt(ciphertext)
+
+        bytes_original_capsule = capsule._original_to_bytes()
+        cleartext = dem.decrypt(ciphertext, authenticated_data=bytes_original_capsule)
     else:
         key = _decapsulate_original(priv_key.bn_key, capsule)
         dem = UmbralDEM(key)
-        cleartext = dem.decrypt(ciphertext)
+
+        bytes_capsule = bytes(capsule)
```
Comment (jMyles) [inline umbral/pre.py]: Usually, we've been saying `<entity_type>_bytes` to name a bytes representation of something (ie, `capsule_bytes`, not `bytes_capsule`)

File: umbral/fragments.py [hunk 6]
```diff
@@ -53,31 +53,20 @@ def to_bytes(self):
...
-
-        h = params.h
-        lh_exp = self.bn_key * h
-
-        rh_exp = vKeys[0]
-        i_j = self.bn_id
-        for vKey in vKeys[1:]:
-            rh_exp = rh_exp + (i_j * vKey)
-            i_j = i_j * self.bn_id
+        check_kfrag_1 = z1 == hash_to_bn([g_y, self.bn_id, pub_a, pub_b, u1, x], params)
```
Comment (jMyles) [inline umbral/fragments.py]: I think I need a comment here.  Why do I expect the hash to be the same as `z1`?

File: tests/test_simple_api.py [hunk 7]
```diff
@@ -31,7 +31,7 @@ def test_simple_api(N, M, curve=default_curve()):
     cleartext = pre.decrypt(capsule, priv_key_alice, ciphertext)
     assert cleartext == plain_data
 
-    rekeys, _unused_vkeys = pre.split_rekey(priv_key_alice, pub_key_bob, M, N, params=params)
+    rekeys = pre.split_rekey(priv_key_alice, pub_key_bob, M, N, params=params)
```
Comment (KPrasch) [inline tests/test_simple_api.py]: looks like `rekeys` == `kfrags`?

File: docs/source/using_pyumbral.rst (hunk 3 above)
Comment (KPrasch) [reply docs/source/using_pyumbral.rst]: Excellent :+1:

File: docs/source/using_pyumbral.rst (hunk 2 above)
Comment (cygnusv) [reply docs/source/using_pyumbral.rst]: Thou shalt not take the name of the Lord thy God in vain

File: umbral/pre.py (hunk 4 above)
Comment (cygnusv) [reply umbral/pre.py]: I don't get your comment ^_^U 
Is that a private joke?

Comment (tuxxy) [reply umbral/pre.py]: It's an allusion to this famous scene from Pulp Fiction.

https://youtu.be/Hrm-rPSCIBw

Comment (cygnusv) [reply umbral/pre.py]: Oh right...but poor Brad ended dead :(

Comment (tuxxy) [reply umbral/pre.py]: They also shot Marvin in the face.. :'(

RIP Marvin

File: umbral/fragments.py (hunk 6 above)
Comment (cygnusv) [reply umbral/fragments.py]: Done. Basically this is just the check performed to validate a Schnorr signature

**Pull request merged**